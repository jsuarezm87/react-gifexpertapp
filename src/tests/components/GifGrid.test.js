import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas en <GifGrid />', () => {

    const category = 'Dragon Ball';
    
    test('debe de mostrarse corrextamente ', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

       const wrapper = shallow( <GifGrid category={category} /> );
       expect( wrapper ).toMatchSnapshot(); 

    });

    test('debe de mostrar items cuando se carga imagenes con useFetchGifs ', () => {

        const gifs = [{
            id: 'ABC',
            url: 'http://localhost/cualquier/cosa.jpg',
            title: 'Cualquier cosa'
        }];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper = shallow( <GifGrid category={category} /> );
        
        expect( wrapper.find('p').exists() ).toBe(false);   
        expect( wrapper.find('GifGridItem').length ).toBe( gifs.length );
        
    });
    
    
});
