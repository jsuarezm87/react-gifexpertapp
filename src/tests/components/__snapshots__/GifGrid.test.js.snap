// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid /> debe de mostrarse corrextamente  1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeInDown"
  >
    Dragon Ball
  </h3>
  <p
    className="animate__animated animate__fadeInDown"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
